### PR TITLE
build: Downgrade patchelf version from 0.18.0 to 0.17.2 

### PR DESCRIPTION
**Pull request by nv-kmcgill53**
#### What does the PR do?

Downgrade patchelf version from 0.18.0 to 0.17.2 due to patchelf regression

Patchelf shipped a regression in 0.18.0 and has since yanked the pypi release pointing to
0.17.2 as the most recent version. However, 0.18.0 is still the version shipped in both the apt and yum
repositories, thus we must use pip to install the version we want.
See https://github.com/mayeut/patchelf-pypi/issues/87

#### Checklist
- [X] PR title reflects the change and is of format `<commit_type>: <Title>`
- [X] Changes are described in the pull request.
- [X] Related issues are referenced.
- [X] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [x] Added [test plan](#test-plan) and verified test passes.
- [x] Verified that the PR passes existing CI.
- [x] Verified copyright is correct on all changed files.
- [x] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [x] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [X] build
- [ ] ci
- [ ] docs
- [ ] feat
- [ ] fix
- [ ] perf
- [ ] refactor
- [ ] revert
- [ ] style
- [ ] test

#### Related PRs:

- https://github.com/triton-inference-server/pytorch_backend/pull/145
- https://github.com/triton-inference-server/openvino_backend/pull/94
- https://github.com/triton-inference-server/onnxruntime_backend/pull/295
- https://github.com/triton-inference-server/tensorflow_backend/pull/109

#### Test plan:

Core `L0_*` tests were segfaulting on loading a shared library used by the pytorch backend. Now they are failing due to known issues with the TensorRT backend.

- CI Pipeline ID:

22812543

#### Background

No further explanation required.

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)
- Relates to https://github.com/mayeut/patchelf-pypi/issues/87

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -1562,7 +1565,10 @@ def add_cpu_libs_to_linux_dockerfile(backends, target_machine):
 
 # patchelf is needed to add deps of libcublasLt.so.12 to libtorch_cuda.so
 RUN apt-get update \\
-      && apt-get install -y --no-install-recommends openmpi-bin patchelf
+      && apt-get install -y --no-install-recommends openmpi-bin 
+RUN pip3 install --upgrade pip \\
+    && pip3 install --upgrade \\
```
Comment (rmccorm4) [inline build.py:1570]: Same as https://github.com/triton-inference-server/server/pull/7959/files#r1925650098

**Review by rmccorm4 — APPROVED** [review]

**Review by rmccorm4 — CHANGES_REQUESTED** [review]
edit - please fix pre-commit failures before merge

File: build.py (hunk 1 above)
Comment (nv-kmcgill53) [reply build.py]: updated.

**Review by rmccorm4 — APPROVED** [review]
